src/mkdocstrings/handlers/rendering.py

```diff
@@ -146,19 +146,35 @@ def __init__(self, md: Markdown, id_prefix: str):
         self.id_prefix = id_prefix
 
     def run(self, root: Element) -> None:  # noqa: D102 (ignore missing docstring)
-        if not self.id_prefix:
-            return
-        for el in root.iter():
-            id_attr = el.get("id")
-            if id_attr:
-                el.set("id", self.id_prefix + id_attr)
+        if self.id_prefix:
+            self._prefix_ids(root)
 
+    def _prefix_ids(self, root: Element) -> None:
+        index = -1
+        for el in reversed(root):  # Reversed mainly for the ability to mutate during iteration.
+            index += 1
```

Comment:
The iteration is in reverse order but the indices are in forward order :thinking: so this will fail with multiple elements present

Please consider fetching and merging my fix commit https://github.com/oprypin/mkdocstrings/commit/735e2966db85779cdd459ec6ff6976218d2c0580

Reply:
Of the importance of code reviews... How did I miss that 🫠
Thanks!